src/object/srv_obj_migrate.c

```diff
@@ -996,7 +997,9 @@ __migrate_fetch_update_parity(struct migrate_one *mrone, daos_handle_t oh,
 		for (j = 1; j < iods[i].iod_nr; j++) {
 			daos_recx_t	*recx = &iods[i].iod_recxs[j];
 
-			if (offset + size == recx->rx_idx) {
+			/* Merge the recx if there are in the same stripe */
+			if (offset + size == recx->rx_idx &&
+			    offset / stride_nr == recx->rx_idx / stride_nr) {
```

Comment:
not quite sure about the details here, just confirm the "stride_nr" here should use "obj_ec_stripe_rec_nr(&mrone->mo_oca)" rather than "obj_ec_cell_rec_nr(...)"?